src/components/pages/TicTacToe.js

```diff
@@ -55,39 +51,130 @@ WinnerCard.propTypes = {
   onRestart: PropTypes.func,
 };
 
-const getWinner = tiles => {
+const getWinner = (tiles) => {
   // calcular el ganador del partido a partir del estado del tablero
   // (existen varias formas de calcular esto, una posible es listar todos los
   // casos en los que un jugador gana y ver si alguno sucede)
+  // Ain't nobody got time for that
+  if (tiles[0] !== '' && tiles[0] === tiles[1] && tiles[1] === tiles[2]) {
+    return tiles[0];
+  }
+  if (tiles[3] !== '' && tiles[3] === tiles[4] && tiles[4] === tiles[5]) {
+    return tiles[3];
+  }
+  if (tiles[6] !== '' && tiles[6] === tiles[7] && tiles[7] === tiles[8]) {
+    return tiles[6];
+  }
+  if (tiles[0] !== '' && tiles[0] === tiles[3] && tiles[3] === tiles[6]) {
+    return tiles[0];
+  }
+  if (tiles[1] !== '' && tiles[1] === tiles[4] && tiles[4] === tiles[7]) {
+    return tiles[1];
+  }
+  if (tiles[2] !== '' && tiles[2] === tiles[5] && tiles[5] === tiles[8]) {
+    return tiles[2];
+  }
+  if (tiles[0] !== '' && tiles[0] === tiles[4] && tiles[4] === tiles[8]) {
+    return tiles[0];
+  }
+  if (tiles[2] !== '' && tiles[2] === tiles[4] && tiles[4] === tiles[6]) {
+    return tiles[2];
+  }
   return null;
```

Comment:
calculo que esto lo hiciste para tratar de hacer que el algoritmo fuera O(1) por lo que comentaste en slack. De todas formas como estás haciendo muchas comparaciones, es lo mismo que meter todo esto en un `for`. Si podés te recomendaría hacerlo de esa forma.
Si seguís empeñado en hacerlo con complejidad baja como comentaste, *CREO* que se me ocurre una forma de hacerlo en O(log n), pero no creo que puedas llegar a O(1)